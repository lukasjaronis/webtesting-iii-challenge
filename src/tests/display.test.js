import React from "react";
import { render } from "@testing-library/react";
import { closed, locked } from '../controls/Controls';

import Display from '../display/Display';

// closed
test('Matches closed gate snapshot', () => {
const closedClass = `led ${closed ? 'red-led' : 'green-led'}`;
const tree = render(<Display className={closedClass} />)
expect(tree.baseElement).toMatchSnapshot();
})

// locked
test('Matches locked gate snapshot', () => {
    const lockedClass = `led ${locked ? 'red-led' : 'green-led'}`;
    const tree = render(<Display className={lockedClass} />)
    expect(tree.baseElement).toMatchSnapshot(); 
})

test('unlocked and open default', () => {
    const defaultState = {
        closed: false,
        locked: false
    }

    const { getByText } = render(<Display closed={defaultState.closed} locked={defaultState.locked} />)

    expect(getByText(/unlocked/i)).toBeDefined();
    expect(getByText(/open/i)).toBeDefined();
})

test('displays if gate is open/closed and locked/unlocked', () => {
    const defaultState = {
        closed: true,
        locked: true
    }

    const { getByText } = render(<Display closed={defaultState.closed} locked={defaultState.locked} />)

    expect(getByText(defaultState.locked ? /locked/i : /unlocked/i)).toBeDefined();
    expect(getByText(defaultState.closed ? /closed/i : /open/i)).toBeDefined();
})