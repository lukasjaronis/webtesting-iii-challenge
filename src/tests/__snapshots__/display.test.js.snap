// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches closed gate snapshot 1`] = `
<body>
  <div>
    <div
      class="display panel"
    >
      <div
        class="led green-led"
      >
        Unlocked
      </div>
      <div
        class="led green-led"
      >
        Open
      </div>
    </div>
  </div>
</body>
`;

exports[`Matches locked gate snapshot 1`] = `
<body>
  <div>
    <div
      class="display panel"
    >
      <div
        class="led green-led"
      >
        Unlocked
      </div>
      <div
        class="led green-led"
      >
        Open
      </div>
    </div>
  </div>
</body>
`;

exports[`display closed if closed prop is true 1`] = `
<body>
  <div>
    <div
      class="display panel"
    >
      <div
        class="led green-led"
      >
        Unlocked
      </div>
      <div
        class="led red-led"
      >
        Closed
      </div>
    </div>
  </div>
</body>
`;
